// @pure0cd/freefire-api/lib/constants.js
module.exports = {
    AE: {
        MAIN_KEY: Buffer.from("Yg&tc%DEuh6%Zc^8", "binary"),
        MAIN_IV: Buffer.from("6oyZDr22E3ychjM%", "binary")
    },
    // -------------------------------------------------------------------------------------------------
    HEADERS: {
        COMMON: {
            'User-Agent': "Dalvik/2.1.0 (Linux; U; Android 13; A063 Build/TKQ1.221220.001)",
            'Connection': "Keep-Alive",
            'Accept-Encoding': "gzip",
            'Expect': "100-continue",
            'X-Unity-Version': "2018.4.11f1",
            'X-GA': "v1 1",
            'ReleaseVersion': "OB52", // Update this when Free Fire updates (e.g., OB53, OB54, etc.). Report to @pure0cd via GitHub Issues.
            'Content-Type': "application/x-www-form-urlencoded"
        },
        GARENA_AUTH: {
            'User-Agent': "GarenaMSDK/4.0.19P9(A063 ;Android 13;en;IN;)",
            'Connection': "Keep-Alive",
            'Accept-Encoding': "gzip"
        }
    },
    // -------------------------------------------------------------------------------------------------
    URLS: {
        GARENA_TOKEN: "https://ffmconnect.live.gop.garenanow.com/oauth/guest/token/grant",
        MAJOR_LOGIN: "https://loginbp.ggblueshark.com/MajorLogin",
        SEARCH: (serverUrl) => `${serverUrl}/FuzzySearchAccountByName`,
        PERSONAL_SHOW: (serverUrl) => `${serverUrl}/GetPlayerPersonalShow`,
        PLAYER_STATS: (serverUrl) => `${serverUrl}/GetPlayerStats`,
        PLAYER_CS_STATS: (serverUrl) => `${serverUrl}/GetPlayerTCStats`
    },
    GARENA_CLIENT: {
        CLIENT_ID: "100067",
        CLIENT_SECRET: "2ee44819e9b4598845141067b281621874d0d5d7af9d8f7e00c1e54715b7d1e3"
    },
    DEFAULT_CREDENTIALS: {
        UID: "3301269321",
        PASSWORD: "D11732AC9BBED0DED65D0FED7728CA8DFF408E174202ECF1939E328EA3E94356"
    }
};
